src/all/mangaball/src/eu/kanade/tachiyomi/extension/all/mangaball/MangaBall.kt

```diff
@@ -0,0 +1,409 @@
+        throw Exception("Unsupported url")
+    }
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        return GET(getMangaUrl(manga), headers)
+    }
+
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl/title-detail/${manga.url}/"
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val document = response.asJsoup()
+        getCSRF(document)
+
+        return SManga.create().apply {
+            url = document.location().toHttpUrl().pathSegments[1]
+            title = document.selectFirst("#comicDetail h6")!!.ownText()
+            thumbnail_url = document.selectFirst("img.featured-cover")?.absUrl("src")
+            genre = buildList {
+                document.selectFirst("#featuredComicsCarousel img[src*=/flags/]")
+                    ?.attr("src")?.also {
+                        when {
+                            it.contains("jp") -> add("Manga")
+                            it.contains("kr") -> add("Manhwa")
+                            it.contains("cn") -> add("Manhua")
+                        }
+                    }
+                document.select("#comicDetail span[data-tag-id]")
+                    .mapTo(this) { it.ownText() }
+            }.joinToString()
+            author = document.select("#comicDetail span[data-person-id]")
+                .eachText().joinToString()
+            description = buildString {
+                document.selectFirst("#descriptionContent p")
+                    ?.also { append(it.wholeText()) }
+                document.selectFirst("#comicDetail span.badge:contains(Published)")
+                    ?.also { append("\n\n", it.text()) }
+                val titles = document.select("div.alternate-name-container").text().split("/")
+                if (titles.isNotEmpty()) {
+                    append("\n\nAlternative Names: \n")
+                    titles.forEach {
+                        append("- ", it.trim(), "\n")
+                    }
+                }
+            }.trim()
+            status = when (document.selectFirst("span.badge-status")?.text()) {
+                "Ongoing" -> SManga.ONGOING
+                "Completed" -> SManga.COMPLETED
+                "Hiatus" -> SManga.ON_HIATUS
+                "Cancelled" -> SManga.CANCELLED
+                else -> SManga.UNKNOWN
+            }
+        }
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val id = manga.url.substringAfterLast("-")
+        val body = FormBody.Builder()
+            .add("title_id", id)
+            .build()
+
+        return POST("$baseUrl/api/v1/chapter/chapter-listing-by-title-id/", headers, body)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        (response.request.body as FormBody).also {
+            updateViews(it.value(0))
+        }
+
+        val data = response.parseAs<ChapterListResponse>()
+        val numberFormatter = DecimalFormat("#.##")
```

Comment:
Usually you would use something like `num.toString().removeSuffix(".0")`, unless you really want to round numbers such as 1.333 to 1.33